### ISSUE TITLE
Support non-block cursor types in integrated terminal

### ISSUE THREAD

**Issue by garfieldbanks** [Microsoft/vscode#10528]
I dislike how the block cursor hides characters when cursorBlinking is set to false, but the blinking drives me crazy. This is only setting right now:  

```
// Controls whether the terminal cursor blinks.  
"terminal.integrated.cursorBlinking": false,  
```

I would appreciate it if you could add these:  

```
// Controls the terminal cursor animation style, possible values are 'blink' and 'solid'  
"terminal.integrated..cursorBlinking": "solid",  

// Controls the terminal cursor style, accepted values are 'block', 'line' and 'underline'  
"terminal.integrated..cursorStyle": "line",
```

**Comment by Tyriar** [comment]
That was a bug that slipped through and is fixed in the Insiders build now (#10100). Let's track cursor style in this issue.

**Comment by Tyriar** [comment]
To verifiers: please ensure `terminal.integrated.cursorStyle` works as expected and is consistent with `editor.cursorStyle`.
